test/unit/hash_for_test.rb

```diff
@@ -0,0 +1,18 @@
+require 'test_helper'
+
+class HashForTest < ActionView::TestCase
+  test "hash_for_*_path returns expected elements" do
+    assert_equal({:controller => 'hosts', :action => 'new', :use_route => 'new_host'}, hash_for_new_host_path)
+  end
+
+  test "hash_for_* doesn't memorize options" do
```

Comment:
This is a test for the pre-existing `opts.dup` line, explained at https://github.com/theforeman/foreman/pull/3294/files#r55810116.